Added an error event publisher

hitohata opened the pull request:
## Overviews of implementation

* Added an error event publisher
* add it to the controller
* Added a new policy to call EventBridge

## Review points

None

hitohata committed:
feat: implemented an error event publisher

taniguchiiqqq commented on cdk/lib/nishikiBackend-stack.ts:
```diff
@@ -179,5 +179,14 @@ const nishikiMainBackendFunction = (
+		new iam.PolicyStatement({
+			actions: ["events:PutEvents"],
+			resources: [
+				`arn:aws:events:${this.region}:${this.account}:event-bus/NishikiNotificationBus`,
```
This line causes an error when it's compiled.
```
TSError: ⨯ Unable to compile TypeScript:
lib/nishikiBackend-stack.ts:186:23 - error TS2532: Object is possibly 'undefined'.

186     `arn:aws:events:${this.region}:${this.account}:event-bus/NishikiNotificationBus`,
```

So instead of `this`, I think scope should be here.
```typescript
const nishikiMainBackendFunction = (
	scope: cdk.Stack,
```

```suggestion
				`arn:aws:events:${scope.region}:${scope.account}:event-bus/NishikiNotificationBus`,
```

taniguchiiqqq reviewed (changes requested):
I'm so sorry for late review. This changes must be helpful to find abnormal situations quickly. 

I found this code can't be deployed directly because of this error. I'm not sure why this change requires deleting UserPool yet.
```
Export NishikiStaticAssetsStack-dev:ExportsOutputRefNishikiUserPool1CB49506EF32CA86 cannot be deleted as it is in use by NishikiBackendStack-dev
```

hitohata committed:
chore: merged the develop branch

hitohata committed:
fix: account and region settings

hitohata committed:
fix: package-lock.json

hitohata commented:
@taniguchiiqqq 
Thank you for your reviewing, I have updated the region and account. (the Construct object doesn't have a region and account property so I changed it a little.)

> I found this code can't be deployed directly because of this error. I'm not sure why this change requires deleting UserPool yet.

This behavior didn't occur in my environment.

taniguchiiqqq reviewed (approved):
Thanks! LGTM. It looks better to pass region and account than pass a whole stack.
